service/dap/server_test.go

```diff
@@ -2905,6 +2915,219 @@ func TestUnupportedCommandResponses(t *testing.T) {
+					client.SetVariableRequest(barRef, "Baz", "42")
+					client.ExpectStoppedEvent(t)
+					if got, want := client.ExpectSetVariableResponse(t), "42"; got.Success != true || got.Body.Value != want {
+						t.Errorf("got %#v, want {Success=true, Body.Value=%q", got, want)
+					}
+					handleStop(t, client, 1, "main.foobar", startLineno)
+
+					client.VariablesRequest(1000)
+					args2 := client.ExpectVariablesResponse(t)
+					expectVarExact(t, args2, 1, "bar", "bar", `main.FooBar {Baz: 42, Bur: "lorem"}`, hasChildren)
+
+					// Test: set integer 'a2'
+					client.VariablesRequest(1001)
+					locals := client.ExpectVariablesResponse(t)
+					expectVarExact(t, locals, -1, "a2", "a2", "6", noChildren)
+					client.SetVariableRequest(1001, "a2", "42")
+					client.ExpectStoppedEvent(t)
+
+					if got, want := client.ExpectSetVariableResponse(t), "42"; got.Success != true || got.Body.Value != want {
+						t.Errorf("got %#v, want {Success=true, Body.Value=%q}", got, want)
+					}
+
+					handleStop(t, client, 1, "main.foobar", startLineno)
+
+					client.VariablesRequest(1001)
+					locals2 := client.ExpectVariablesResponse(t)
+					expectVarExact(t, locals2, -1, "a2", "a2", "42", noChildren)
+
+					// Test: set integer 'a2' with an invalid value.
+					client.SetVariableRequest(1001, "a2", "false")
+					if got, want := client.ExpectErrorResponse(t), "can not convert false constant to int"; !strings.Contains(got.Body.Error.Format, want) {
+						t.Errorf("got %#v, want error string containing %q", got, want)
+					}
+
+					// Test: set global integer 'p1'.
+					client.VariablesRequest(1002)
+					globals := client.ExpectVariablesResponse(t)
+					expectVarExact(t, globals, -1, "p1", "main.p1", "10", noChildren)
+					client.SetVariableRequest(1002, "p1", "-10")
+					client.ExpectStoppedEvent(t)
+					if got, want := client.ExpectSetVariableResponse(t), "-10"; got.Success != true || got.Body.Value != want {
+						t.Errorf("got %#v, want {Success=true, Body.Value=%q", got, want)
+					}
+
+					handleStop(t, client, 1, "main.foobar", startLineno)
+
+					client.VariablesRequest(1002)
+					globals2 := client.ExpectVariablesResponse(t)
+					expectVarExact(t, globals2, -1, "p1", "main.p1", "-10", noChildren)
+				},
+				disconnect: true,
+			}})
+	})
+}
+
+// TestSetVariableWithCall tests SetVariable functionality that may require function calls support.
+func TestSetVariableWithCall(t *testing.T) {
+	protest.MustSupportFunctionCalls(t, testBackend)
+	runTest(t, "testvariables", func(client *daptest.Client, fixture protest.Fixture) {
+		runDebugSessionWithBPs(t, client, "launch",
+			func() {
+				client.LaunchRequestWithArgs(map[string]interface{}{
+					"mode": "exec", "program": fixture.Path, "showGlobalVariables": true,
+				})
+			},
+			// Stop at the breakpoints set within the program, and after returning from barfoo (67)
+			fixture.Source, []int{67},
```

Comment:
I thought you removed this? Maybe that was the other test. Also it might be easier to just set 66  explicitly, then to rely on runtime breakpoint and then special case the line number. We did that in some test to test that we stop at correct lines, but you shouldn't need this here.

Reply:
I don't know if understand, but this is just an instruction numbering problem, on windows the first instruction after returning from the call is a NOP that is numbered with the same line as the function call, on other operating systems it's the first instruction of the next line.